### ISSUE TITLE
Groups of settings

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5719]
### Why this would be helpful

There are a few addons with a large number of settings, most of which being basically the same feature but for a different type of whatever, like which kinds of blocks show related blocks in their right click menus or which types of activity you get notified of. It can make the settings a little harder to navigate to have them all stand out and look just as important and separate as every other setting in that addon.

### My suggestion

Mainly for addons like `scratch-notifier` or `block-switching`, it'd help to have similar settings grouped together. This would allow us to make what could be looked at as one big setting with a list of options.

It's slightly hard to explain, especially in the way I see it, so here's an image:
![A mock-up of a settings page for a "push notifications" feature. Below the other on/off switches, a group of settings called "receive notifications for..." contains checkboxes for specific types of activity.](https://user-images.githubusercontent.com/106490990/218284097-510a25f7-1efb-4e61-92e2-705dfb8ae0a3.png)

### Additional context

This won't improve the look of the settings for most addons, but it's worth considering, especially if we plan to redesign the settings page. Given our easy-to-use format, this design improvement is worth it.

**Comment by WorldLanguages** [comment]
This would be interesting for some addons that have too many settings

**Comment by Samq64** [comment]
This could be done with `<fieldset>`.

**Comment by mybearworld** [comment]
It does.
![A demonstration of it working with fieldset in the pop up](https://github.com/ScratchAddons/ScratchAddons/assets/130385691/dff7c4fb-aedd-4555-8ead-7e03fba84e81)

**Comment by DNin01** [comment]
I think option 1 would allow for more flexibility (you can put the groups exactly where you want and the names could easily be localized). I don't know which one would be the easiest to parse or code.

**Comment by CST1229** [comment]
> I don't know which one would be the easiest to parse or code.

Probably 1.

**Comment by WorldLanguages** [comment]
Option 2 requires the least line changes for existing manifests and doesn't add too much complexity to the manifest schema.
But for translation of the addon groups to actually work, we must assign an ID for each setting group and, optionally, provide a name that will be read by the user and that will be sent for translation.

So the final result would be:

```jsonc
// Option 2 - modified:
"settingGroups": [
 {
    "id": "notifyOn",
    "name": "Notify on" // Can be `null`. See #6068
 }
],
"settings": [
  {
    "name": "Loves",
    "id": "loves",
    "type": "boolean",
    "default": true,
    "group": "notifyOn"
  },
  {
    "name": "Favorites",
    "id": "favorites",
    "type": "boolean",
    "default": true,
    "group": "notifyOn"
  },
  // etc
]
```

**Comment by mybearworld** [comment]
That's the best compromise between the two options for sure.

**Comment by DNin01** [comment]
> Option 2 requires the least line changes for existing manifests and doesn't add too much complexity to the manifest schema.

Yeah. That way Scratch Addons doesn't have to check whether what it's looking at is a group or a setting.
